v2/internal/controllers/crd_dataprotection_backuppolicy_test.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	// The testing package is imported for testing-related functionality.
+	"testing"
+
+	// The gomega package is used for assertions and expectations in tests.
+	. "github.com/onsi/gomega"
+
+	// The dataprotection package contains types and functions related to dataprotection resources.
+	dataprotection "github.com/Azure/azure-service-operator/v2/api/dataprotection/v1api20230101"
+	// The testcommon package includes common testing utilities.
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+)
+
+func Test_Dataprotection_Backuppolicy_CRUD(t *testing.T) {
+	// indicates that this test function can run in parallel with other tests
+	t.Parallel()
+
+	// tc := globalTestContext.ForTest(t) initializes the test context for this test.
+	// The globalTestContext is a global object that provides the necessary context and utilities for testing.
+	tc := globalTestContext.ForTest(t)
+
+	// rg := tc.CreateTestResourceGroupAndWait() creates a test resource group and waits until the operation is completed.
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	// Consts for BackupVault
+	identityType := "SystemAssigned"
+	alertsForAllJobFailures_Status := dataprotection.AzureMonitorAlertSettings_AlertsForAllJobFailures_Enabled
+	StorageSetting_DatastoreType_Value := dataprotection.StorageSetting_DatastoreType_VaultStore
+	StorageSetting_Type_Value := dataprotection.StorageSetting_Type_LocallyRedundant
+
+	// Create a backupvault
+	backupvault := &dataprotection.BackupVault{
+		ObjectMeta: tc.MakeObjectMetaWithName("shayvault1"),
+		Spec: dataprotection.BackupVault_Spec{
+			Location: tc.AzureRegion,
+			Tags:     map[string]string{"cheese": "blue"},
+			Owner:    testcommon.AsOwner(rg),
+			Identity: &dataprotection.DppIdentityDetails{
+				Type: &identityType,
+			},
+			Properties: &dataprotection.BackupVaultSpec{
+				MonitoringSettings: &dataprotection.MonitoringSettings{
+					AzureMonitorAlertSettings: &dataprotection.AzureMonitorAlertSettings{
+						AlertsForAllJobFailures: &alertsForAllJobFailures_Status,
+					},
+				},
+				StorageSettings: []dataprotection.StorageSetting{
+					{
+						DatastoreType: &StorageSetting_DatastoreType_Value,
```

Comment:
```suggestion
						DatastoreType: to.Ptr(dataprotection.StorageSetting_DatastoreType_VaultStore),
```

Consider using `to.Ptr()` for all the enum values above and below, rather than assigning to a var.

Reply:
Thanks for mentioning the library. That helped to reduce the code length.
made the possible change sin both BackupVault and BackupPolicy files.